File: 13. Roman to Integer/solution.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function romanToInt($s) {
        $rule = [
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000,
        ];
        $sum = 0;

        for ($i=0;$i<strlen($s);$i++) {
            if ($rule[$s[$i]] < $rule[$s[$i+1]]) {
                $sum -= $rule[$s[$i]];
            } else {
                $sum += $rule[$s[$i]];
            }
        }

        return $sum;
    }
}

File: 13. Roman to Integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanDict = {
            'M': 1000,
            'D': 500,
            'C': 100,
            'L': 50,
            'X': 10,
            'V': 5,
            'I': 1
        }
        result = 0
        for i in range(len(s) - 1):
            if romanDict[s[i]] < romanDict[s[i+1]]:
                result -= romanDict[s[i]]
            else:
                result += romanDict[s[i]]
        result += romanDict[s[-1]]
        return result